File: src/main/java/imaxct/service/ICourseService.java
package imaxct.service;

import imaxct.domain.Course;

import java.util.List;

/**
 * Created by maxct on 2017/4/9.
 */
public interface ICourseService {
    boolean addCourse(Course course);

    boolean deleteCourse(Course course);

    /**
     * id : -1 all
     * */
    List<Course> getAllCourses(int id);

    List<Course> getCourseByName(String name);
}

File: src/main/java/imaxct/service/ICourseService.kt
package imaxct.service

import imaxct.domain.Course

/**
 * Created by maxct on 2017/4/9.
 */
interface ICourseService {
    fun addCourse(course: Course): Boolean

    fun deleteCourse(course: Course): Boolean

    /**
     * id : -1 all
     */
    fun getAllCourses(id: Int): List<Course>

}
